### ISSUE TITLE
Darglint + Class Docs

### ISSUE THREAD

**Issue by adithyabsk** [terrencepreilly/darglint#25]
It seems that darglint currently does not support the Google style documentation of [documenting class params in the class documentation](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) rather than the \_\_init__. 

Is this planned to be fixed? Not sure how technically feasible this would be based on the conversation from #21.

E.g.
```python
class Test(object):
"""Tests darglint.

Params:
    p1 (int): param 1
    p2 (str): param 2

"""

    def __init__(self, p1, p2):
        self.p1 = p1
        self.p2 = p2
```

**Comment by terrencepreilly** [comment]
I've considered this before.  Parsing the class docstring is no problem, but verifying the attributes of the class does provide some challenges.  The problem is that the attributes visible in the interface could be resolved in many different ways.  They could be instantiated directly on the class, in a function (not limited to `__init__`), *as* a function (getters and setters), or appended directly to the namespace through reflection.

This feature wouldn't be able to cover all possibilities, but it seems like it would be useful for darglint to check for attributes instantiated either on the class or in the `__init__` method.  I'll try to get to this after handling style errors. (I'm about 30 commits into it and hopefully getting close to a solution.)

**Comment by adithyabsk** [comment]
That sounds fantastic, thank you for your hard work on this package. This issue is on the minimum pathway to get darglint integrated into our tooling setup and I'll keep an eye on this issue until then 😃

**Comment by sobolevn** [comment]
Would love to have this feature!

**Comment by Beefy-Swain** [comment]
Is this related to the following issue

```python
class Color(NamedTuple):
    """Color object.

    Attributes:
        green (int): Green, between 1 and 255.
        blue (int): Blue, between 1 and 255.
        alpha (int): Alpha, between 1 and 255.
    """

    red: int
    green: int
    blue: int
    alpha: int```

not throwing up an error relating to `red` not being documented?

**Comment by terrencepreilly** [comment]
@Beefy-Swain Correct; this is not yet a feature of darglint. (Because of the difficulties in statically analyzing attributes.)

**Comment by sobolevn** [comment]
Maybe we can stick to this algorithm?

1. If there's an explicit class attribute - it has to be documented 
2. If there's an extra attribute in the class docs, that's fine (because we cannot know the full list of arguments in this time)

**Comment by adithyabsk** [comment]
@terrencepreilly Is this issue still on the roadmap?

**Comment by terrencepreilly** [comment]
Yes, I just have limited bandwidth at the moment.  I plan on getting to it at some point.  I've updated the [roadmap](https://github.com/terrencepreilly/darglint/tree/develop#roadmap) in develop to include it.

**Comment by adithyabsk** [comment]
@terrencepreilly got it, thanks for the update
